sansa-inference-spark/src/main/scala/net/sansa_stack/inference/spark/forwardchaining/axioms/ForwardRuleReasonerOWLHorst.scala

```diff
@@ -0,0 +1,650 @@
+package net.sansa_stack.inference.spark.forwardchaining.axioms
+
+import java.io.File
+
+import net.sansa_stack.inference.utils.CollectionUtils
+import net.sansa_stack.owl.spark.rdd.{FunctionalSyntaxOWLAxiomsRDDBuilder, OWLAxiomsRDD}
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.{RDD, UnionRDD}
+import org.apache.spark.sql.SparkSession
+import org.semanticweb.owlapi.apibinding.OWLManager
+import org.semanticweb.owlapi.model._
+
+import scala.collection.JavaConverters._
+
+
+class ForwardRuleReasonerOWLHorst (sc: SparkContext, parallelism: Int = 2) extends TransitiveReasoner{
+
+  def this(sc: SparkContext) = this(sc, sc.defaultParallelism)
+
+  def apply(axioms: RDD[OWLAxiom], input: String): RDD[OWLAxiom] = {
```

Comment:
For testability it would be better to break this long method down to smaller protected methods that can then be tested individually.